server/src/main/java/org/opensearch/index/translog/transfer/FileSnapshot.java

```diff
@@ -46,6 +49,14 @@ private FileSnapshot(Path path) throws IOException {
         this.fileChannel = FileChannel.open(path, StandardOpenOption.READ);
     }
 
+    private FileSnapshot(Path path, Map<String, String> metadata) throws IOException {
```

Comment:
can we have just one ctor initialising fields and rest all other ctor using `this(...)`?

Reply:
I would suggest to minimise number constructors in general. Lets also try reducing ctors for TranslogFileSnapshot and CheckpointFileSnapshot.

Reply:
@skumawat2025  this has not been resolved yet?